brainstormforce/astra
# refactor: Improved performance by using async functions with promises

### Description
- As we need to register settings and controls initially before using, it's usually taking a time to loop over thousands of array of the objects which blocking a customizer page to load.
- Made function asynchronous and wait for the promise to complete which will not wait for the entire loop and dependent function will execute after the async execution.

### Screenshots
NA

### Types of changes
- Made possible functions asynchronous.
- Added dependent functions inside promises. 

### How has this been tested?
- Check the performance benefit by comparing the previous version.
- Check possible actions like partial refresh, tab switch, grid drop, etc.

### Checklist:
